src/Compilers/CSharp/Portable/Syntax/Syntax.xml

```diff
@@ -3391,6 +3391,7 @@
     <Kind Name="SimpleBaseType"/>
     <Field Name="Type" Type="TypeSyntax" Override="true">
     </Field>
+    <Field Name="ArgumentList" Type="ArgumentListSyntax" Optional="true"/>
```

Comment:
>ArgumentList [](start = 17, length = 12)

It feels like this is a case where a different node derived from `BaseTypeSyntax` would make more sense than adding arguments to `SimpleBaseTypeSyntax`

Reply:
> It feels like this is a case where a different node derived from BaseTypeSyntax would make more sense than adding arguments to SimpleBaseTypeSyntax

I considered that, however I believe existing tools will be served better by not forcing them to change what types of nodes they need to handle. I'll record this as an open issue. 

---
In reply to: [433449588](https://github.com/dotnet/roslyn/pull/44733#discussion_r433449588) [](ancestors = 433449588)

Reply:
This anticipated future scenario (adding arguments in the base clause for records) is precisely the reason that `BaseTypeSyntax` was separated from `SimpleBaseTypeSyntax` in the first place.

---
In reply to: [433461884](https://github.com/dotnet/roslyn/pull/44733#discussion_r433461884) [](ancestors = 433461884,433449588)

Reply:
> This anticipated future scenario (adding arguments in the base clause for records) is precisely the reason that BaseTypeSyntax was separated from SimpleBaseTypeSyntax in the first place.

Quite possibly true. We didn't have that many existing tools at the time. In order to avoid a possibility that many of the tools would need an adjustment to deal with the new node, I believe it is safer to not introduce a new node for preview 3. We will follow up on this design question at a later time. 

---
In reply to: [433525828](https://github.com/dotnet/roslyn/pull/44733#discussion_r433525828) [](ancestors = 433525828,433461884,433449588)